src/commands/registries/azure/DockerAssignAcrPullRoleStep.ts

```diff
@@ -0,0 +1,56 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Progress } from "vscode";
+import { IAppServiceWizardContext } from "vscode-azureappservice"; // These are only dev-time imports so don't need to be lazy
+import { AzureWizardExecuteStep, createAzureClient } from "vscode-azureextensionui";
+import { ext } from "../../../extensionVariables";
+import { localize } from "../../../localize";
+import { AzureRegistryTreeItem } from '../../../tree/registries/azure/AzureRegistryTreeItem';
+import { RemoteTagTreeItem } from '../../../tree/registries/RemoteTagTreeItem';
+
+export class DockerAssignAcrPullRoleStep extends AzureWizardExecuteStep<IAppServiceWizardContext> {
+    public priority: number = 141; // execute after DockerSiteCreateStep
+
+    public constructor(private readonly tagTreeItem: RemoteTagTreeItem) {
+        super();
+    }
+
+    public async execute(context: IAppServiceWizardContext, progress: Progress<{ message?: string; increment?: number }>): Promise<void> {
+        const message: string = localize('vscode-docker.commands.registries.azure.deployImage.assigningPullRole', 'Granting permission for App Service to pull image from ACR...');
+        ext.outputChannel.appendLine(message);
+        progress.report({ message: message });
+
+        const armAuth = await import('@azure/arm-authorization');
+        const armContainerRegistry = await import('@azure/arm-containerregistry');
+        const armAppService = await import('@azure/arm-appservice');
+        const authClient = createAzureClient(context, armAuth.AuthorizationManagementClient);
+        const crmClient = createAzureClient(context, armContainerRegistry.ContainerRegistryManagementClient);
+        const appSvcClient = createAzureClient(context, armAppService.WebSiteManagementClient);
+
+        // If we're in `execute`, then `shouldExecute` passed and `this.tagTreeItem.parent.parent` is guaranteed to be an AzureRegistryTreeItem
+        const registryTreeItem: AzureRegistryTreeItem = this.tagTreeItem.parent.parent as AzureRegistryTreeItem;
+
+        // 1. Get the registry resource. We will need the ID.
+        const registry = await crmClient.registries.get(registryTreeItem.resourceGroup, registryTreeItem.registryName);

```

Comment:
As discussed via chat, consider adding some error handling if one of these Azure REST calls fails